### DISCUSSION TITLE
Web-Based UI and Server-Side Conversion for UVR with Docker Integration

### DISCUSSION THREAD

**Discussion by remyayad** [Anjok07/ultimatevocalremovergui#1584] [Ideas]
It would be fantastic to develop and deploy a web-based UI for UVR, containerized with Docker, to allow hosting the application on a server (such as a homelab or other setup) which generally has more resources than typical end-user machines. This way, users could upload their tracks via the web interface, and the conversion would take place on the server. Once the process is complete, the 'converted' file could be easily downloaded back to their local machine. What are your thoughts?

**Comment by deepfriedheroin** [comment]
there are a whole bunch of these https://github.com/Eddycrack864/UVR5-UI. once you look at it, you will get an idea of what goes inside and it should make a lot more sense, as to how you can roll your own, easily, if any existing solutions don't work for you.

**Reply by remyayad** [reply]
These tools are somewhat limited in terms of resources usage. You must use very low settings to successfully complete a conversion within the GPU’s allowable usage limits. An entirely self-hosted solution would be ideal.
